src/controller_examples/zookeeper_controller/proof/liveness.rs

```diff
@@ -1192,17 +897,113 @@ proof fn lemma_true_leads_to_always_current_state_matches_zk_under_eventual_inva
         spec,
         true_pred(),
         lift_state(|s: ClusterState| !s.reconcile_state_contains(zk.object_ref())),
-        lift_state(at_init_step_with_no_pending_req(zk)),
-        lift_state(at_after_create_headless_service_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_create_client_service_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_create_admin_server_service_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_create_config_map_step_with_zk_and_pending_req_in_flight(zk)),
-        lift_state(at_after_get_stateful_set_step_with_zk_and_pending_req_in_flight(zk)),
+        lift_state(no_pending_req_at_zookeeper_step_with_zk(zk, ZookeeperReconcileStep::Init)),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateHeadlessService, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateClientService, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateAdminServerService, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterCreateConfigMap, zk, arbitrary())),
+        lift_state(pending_req_in_flight_at_zookeeper_step_with_zk(ZookeeperReconcileStep::AfterGetStatefulSet, zk, arbitrary())),
         lift_state(current_state_matches(zk)),
         always(lift_state(current_state_matches(zk)))
     );
 }
 
+proof fn lemma_from_pending_req_in_flight_at_some_step_to_next_step(
```

Comment:
This is nice!

Reply:
I am even thinking about whether we can further generalize this lemma (make it applicable to all controllers), but let's leave it for future PRs

Reply:
Could you document this lemma? Especially why we need these preconditions. The documentation will help us later a lot.

Reply:
> I am even thinking about whether we can further generalize this lemma

This is what we need to do later.

Reply:
And generalizing it seems not difficult but requires more handlings of "correct requests".